chanzuckerberg/idseq-web
# [Data Discovery] Fix data loading issues

* Fix issues where sidebar was not showing filtered data on: project view, string search
* Refactored code a bit to make data requests more clear
* Add description of data retrieval workflow to make it easier to understand

Testing:
- click "my data" and check sidebar numbers to see if they make sense (harder to see in public page)
- choose a project and check sidebar numbers to see if they make sense (harder to see in public page)
- within projects, samples and single project page combine with some filters, search string on top and check that metadata counts vary in a consistent manner. 
- for checking values, compare metadata counts with the information displayed on the table
